hbase-client/src/main/java/org/apache/hadoop/hbase/client/RegionServerCallable.java

```diff
@@ -73,18 +75,19 @@ public abstract class RegionServerCallable<T, S> implements RetryingCallable<T>
    * @param row           The row we want in <code>tableName</code>.
    */
   public RegionServerCallable(Connection connection, TableName tableName, byte[] row,
-    RpcController rpcController) {
-    this(connection, tableName, row, rpcController, HConstants.NORMAL_QOS);
+    RpcController rpcController, Map<String, byte[]> requestAttributes) {
+    this(connection, tableName, row, rpcController, HConstants.NORMAL_QOS, requestAttributes);
   }
 
   public RegionServerCallable(Connection connection, TableName tableName, byte[] row,
-    RpcController rpcController, int priority) {
+    RpcController rpcController, int priority, Map<String, byte[]> requestAttributes) {
     super();
     this.connection = connection;
     this.tableName = tableName;
     this.row = row;
     this.rpcController = rpcController;
     this.priority = priority;
+    this.requestAttributes = requestAttributes;
```

Comment:
you need to actually use this now, below in `call`. Like where they are setting all the stuff onto hrc, need another `hrc.setRequestAttributes(requestAttributes)` call I think.

I'm surprised the tests are passing (i guess they havent finished yet, so we'll see)